src/utils.ts

```diff
@@ -71,12 +75,16 @@ export async function restoreCaches(cacheDir: string): Promise<void> {
 /**
  * Uploads the cache to GitHub Actions cache from the given path.
  * @param cacheDir The path to upload the cache from.
+ * @param additionalCacheHash Addition to the generated cache hash
  */
-export async function uploadCaches(cacheDir: string): Promise<void> {
+export async function uploadCaches(
+  cacheDir: string,
+  additionalCacheHash: string
+): Promise<void> {
   try {
     await cache.saveCache(
       [cacheDir],
-      `${process.env['RUNNER_OS']}-qodana-${process.env['GITHUB_REF']}-${process.env['GITHUB_SHA']}`
+      `${process.env['RUNNER_OS']}-qodana-${process.env['GITHUB_REF']}-${additionalCacheHash}`
```

Comment:
The previous line didn't make any sense to me as restore, and upload key was not same

Reply:
Cool idea. Was thinking of this too.
But how about we have `process.env['GITHUB_SHA']`by default (or any other key) and then we use custom keys if they were provided

Reply:
You mean setting the default of ` additional-cache-hash` to `${{ github.sha }}`?

Reply:
Yes, that will do

Reply:
Don't forget to bump the package.json version to 4.2.0

Reply:
done